docs/library/ubluetooth.rst

```diff
@@ -73,6 +73,11 @@ Configuration
       example scan results) and the ability for a central role to receive
       larger characteristic values.
 
+    - ``'mtu'``: Get/set the MTU that will be used when a remote device initiates
+      an MTU exchange. The resulting MTU will be the minimum of this and the
```

Comment:
The primary purpose of this setting is for when the remote device initiates the exchange. If this (local) device is the initiator  then the user will do so by calling gattc_exchange_mtu which takes an `mtu` argument.

Admittedly, that `mtu` argument is optional (in which case it defaults to the preferred MTU set here), but the intent was that was a special case for the behavior of gattc_exchange_mtu rather than the intent of setting the preferred mtu.  (i.e. I'm trying to encourage the user that if they're expecting the other side to initiate the exchange, they should use `config(mtu=N)`, and if they're initiating the exchange, they should use `gattc_exchange_mtu(conn_handle, mtu)` ).

Maybe another way to look at this... should `gatt_exchange_mtu()` take an mtu at all, and instead always rely on using the preferred mtu set via `config(mtu=N)`. That would certainly simplify the btstack implementation.

i.e. there are two separate APIs:
 - config --> what do I want the MTU to be (regardless of who the initiator was)
 - gattc_exchange_mtu --> i want to ensure that an exchange takes place

Reply:
Yeah now you say it, I think it's much better to not have the arg option in `gatt_exchange_mtu()`. 
I thought it was a good convenience when I was first coming up with a python api for mtu, but it's really not necessary.

The functionality to allow for the arg is very much non-standard behavior on both stacks with potential side effects and the need for the static variable. It's simpler and more to spec to just split the two api's.

Reply:
I agree that it's simpler to not have any `mtu` argument to `gattc_exchange_mtu()`, and if it doesn't lose any functionality then I'd suggest to remove this arg.  Then it's clear how to set and exchange the MTU, there's only one way to do it.

But, what about specifying a different MTU for each connection?  Is that possible/sensible?

Reply:
> But, what about specifying a different MTU for each connection? Is that possible/sensible?

Yup, that was part of the reason I added it.

I haven't come up with a reason why this would be necessary/useful though. And even if it was, as Andrew says, the stacks don't really work this way. (definitely not btstack, and NimBLE sort of just doesn't prevent it).

I think that makes sense then... I will remove the mtu argument to gattc_exchange_mtu.

Reply:
Each connection will end up with a different mtu if the device on the other end limits it - there shouldn't really be a need for a device to need to pick and choose different settings per connection though.

Reply:
The code would be much simpler with just a global MTU for all connections (because this seems to be the concept that nimble and btstack work on, just a single global MTU).

Reply:
Done. Updated docs/code/tests.